src/gt4py/next/program_processors/runners/dace/gtir_builtin_translators.py

```diff
@@ -576,6 +576,106 @@ def translate_as_fieldop(
+    state: dace.SDFGState,
+    sdfg_builder: gtir_sdfg.SDFGBuilder,
+    domain: gtir.Expr,
+    sym: gtir.Sym,
+    true_br: FieldopData,
+    false_br: FieldopData,
+) -> FieldopData:
+    """
+    Helper function called by `translate_if()` to allocate a temporary field to store
+    the result of an if expression.
+    """
+    assert true_br.gt_type == false_br.gt_type
+    out_type = true_br.gt_type
+
+    if isinstance(sym.type, ts.ScalarType):
+        assert sym.type == out_type
+        dtype = gtx_dace_utils.as_dace_type(sym.type)
+        out, _ = sdfg_builder.add_temp_scalar(sdfg, dtype)
+        out_node = state.add_access(out)
+        return FieldopData(out_node, sym.type, origin=())
+
+    assert isinstance(out_type, ts.FieldType)
+    assert isinstance(sym.type, ts.FieldType)
+    assert sym.type.dims == out_type.dims
+    dims, origin, shape = get_field_layout(extract_domain(domain))
+    assert dims == out_type.dims
+
+    if isinstance(sym.type.dtype, ts.ScalarType):
+        dtype = gtx_dace_utils.as_dace_type(sym.type.dtype)
+    else:
+        assert isinstance(out_type.dtype, ts.ListType)
+        assert isinstance(out_type.dtype.element_type, ts.ScalarType)
+        dtype = gtx_dace_utils.as_dace_type(out_type.dtype.element_type)
+        assert out_type.dtype.offset_type is not None
+        offset_provider_type = sdfg_builder.get_offset_provider_type(
+            out_type.dtype.offset_type.value
+        )
+        assert isinstance(offset_provider_type, gtx_common.NeighborConnectivityType)
+        shape = [*shape, offset_provider_type.max_neighbors]
+
+    out, _ = sdfg_builder.add_temp_array(sdfg, shape, dtype)
+    out_node = state.add_access(out)
+
+    return FieldopData(out_node, out_type, tuple(origin))
+
+
+def _write_if_branch_output(
+    sdfg: dace.SDFG,
+    state: dace.SDFGState,
+    src: FieldopData,
+    dst: FieldopData,
+) -> None:
+    """
+    Helper function called by `translate_if()` to write the result on an if branch,
+    here `src` field, to the output 'dst' field. The write subset is based on the
+    domain of the `dst` field.
+    """
+    if src.gt_type != dst.gt_type:
+        raise ValueError(
+            f"Source and destination type mismatch, '{dst.gt_type}' vs '{src.gt_type}'."
+        )
+    dst_node = state.add_access(dst.dc_node.data)
+    dst_shape = dst_node.desc(sdfg).shape
+
+    if isinstance(src.gt_type, ts.ScalarType):
+        state.add_nedge(
+            src.dc_node,
+            dst_node,
+            dace.Memlet(data=src.dc_node.data, subset="0"),
+        )
+    else:
+        if isinstance(src.gt_type.dtype, ts.ListType):
+            src_origin = [*src.origin, 0]
+            dst_origin = [*dst.origin, 0]
+        else:
+            src_origin = [*src.origin]
+            dst_origin = [*dst.origin]
+
+        data_subset = dace_subsets.Range(
+            (
+                f"{dst_start - src_start}",
+                f"{dst_start + size - src_start - 1}",
```

Comment:
```suggestion
                f"{dst_start - src_start + size - 1}",
```
This makes it more uniform and with the start.
I would also add a comment about why there is a `-1`.